pontoon/base/views.py

```diff
@@ -634,26 +634,66 @@ def get_translation_history(request):
     payload = []
     offset = timezone.now().strftime('%z')
 
+    def get_user_display(user):
+        """Shorcut function that displays user login/credentials or nothing is user is None."""
+        return (user.name_or_email if user else "")
+
     for t in translations:
         u = t.user
-        a = t.approved_user
-        o = {
+        payload.append({
             "id": t.id,
             "user": "Imported" if u is None else u.name_or_email,
             "email": "" if u is None else u.email,
             "translation": t.string,
             "date": t.date.strftime('%b %d, %Y %H:%M'),
             "date_iso": t.date.isoformat() + offset,
             "approved": t.approved,
-            "approved_user": "" if a is None else a.name_or_email,
+            "approved_user": get_user_display(t.approved_user),
+            "unapproved_user": get_user_display(t.unapproved_user),
             "fuzzy": t.fuzzy,
-        }
-        payload.append(o)
+        })
 
     return JsonResponse(payload, safe=False)
 
 
 @require_AJAX
+@login_required
+@transaction.atomic
+def unapprove_translation(request):
+    """Unapprove given translation."""
+    try:
+        t = request.POST['translation']
+        paths = request.POST.getlist('paths[]')
+    except MultiValueDictKeyError as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    translation = Translation.objects.get(pk=t)
+
+    # Only privileged users or authors can un-approve translations
+    if not (request.user.has_perm('base.can_translate_locale', translation.locale)
+            or request.user == translation.user
+            or translation.approved):
+        return HttpResponseForbidden("Forbidden: You can't unapprove this translation.")
+
+    translation.unapprove(request.user)
+    try:
+        previous_translation = translation.entity.translation_set.filter(
+            locale=translation.locale,
+            plural_form=translation.plural_form,
+            pk__lte=translation.pk,
+        )[0].serialize()
+    except IndexError:
+        previous_translation = translation.serialize()
```

Comment:
The idea is to find **the latest translation for the current entity**. It can be the same translation we unapproved, or one of the newer translations.

Something like this:
```
translation = translation.entity.translation_set.filter(
    locale=translation.locale,
    plural_form=translation.plural_form,
).latest('date').serialize()
```